go/src/url_shortener/shorten/server.go

```diff
@@ -0,0 +1,126 @@
+package shorten
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+)
+
+// URLShortener URL shortener server data structure
+type URLShortener struct {
+	ExpanderRoute   string
+	ShortenRoute    string
+	StatisticsRoute string
+
+	Mappings map[string]string
+
+	Statistics StatsJSON
+
+	mux sync.Mutex
+}
```

Comment:
Apart from the mutex, all fields of `URLShortener` are exported. Generally in Go fields are not exported unless being exported brings some value. For example, in an hypothetical `Point{X, Y int} struct` you'd probably want X and Y to be exported in order to access them from outside, etc.

The only reason for which all these fields are exported is because of the lack of proper constructor.